isobus/src/can_internal_control_function.cpp

```diff
@@ -28,79 +26,23 @@ namespace isobus
 	std::shared_ptr<InternalControlFunction> InternalControlFunction::create(NAME desiredName, std::uint8_t preferredAddress, std::uint8_t CANPort)
 	{
 		// Unfortunately, we can't use `std::make_shared` here because the constructor is private
-		auto createdControlFunction = std::shared_ptr<InternalControlFunction>(new InternalControlFunction(desiredName, preferredAddress, CANPort));
-		internalControlFunctionList.push_back(createdControlFunction);
-		return createdControlFunction;
-	}
-
-	bool InternalControlFunction::destroy(std::uint32_t expectedRefCount)
-	{
-		std::unique_lock<std::mutex> lock(controlFunctionProcessingMutex);
-		internalControlFunctionList.erase(std::find(internalControlFunctionList.begin(), internalControlFunctionList.end(), shared_from_this()));
-		lock.unlock();
-
-		return ControlFunction::destroy(expectedRefCount);
-	}
-
-	std::shared_ptr<InternalControlFunction> InternalControlFunction::get_internal_control_function(std::uint32_t index)
```

Comment:
I might have mentioned before, but I think at some point (especially when there's possibly > 1 network manager) I think we will want a way for users to be able to look up their previously created CF objects. Not by index probably, but maybe some other way...? Just thinking out loud, doesn't really affect this PR.

Reply:
Yes, I agree. We expose a const-reference to the vector of CF's I suppose, that maybe is a cleaner approach than limiting to a getter by index?

Reply:
That seems like it would probably be ok? Then they get pretty much do whatever they need to. Doesn't need to affect this PR, just something to keep in mind I guess.

Reply:
I implemented it in commit https://github.com/Open-Agriculture/AgIsoStack-plus-plus/pull/282/commits/d550dc6c7385f6b63ed9925cdada57c0f3a5f332 of this PR while I was at it